temporal-sdk/src/main/java/io/temporal/common/converter/DataConverter.java

```diff
@@ -214,4 +217,25 @@ static Object[] arrayFromPayloads(
       throws DataConverterException {
     return converter.fromPayloads(content, parameterTypes, genericParameterTypes);
   }
+
+  /**
+   * Extract the raw Class from a Type; handles regular classes, parameterized types, and generic
+   * array types.
+   *
+   * @param type the Type to extract from (could be Class, ParameterizedType, or GenericArrayType)
+   * @return the raw Class for the type
+   */
+  static Class<?> getRawClass(Type type) {
```

Comment:
Also don't shouldn't expose this on the data converter interface, this is an implementation detail and should not be part of the public API